File: src/Application Specific/Classes/Entity Collections/ContractManager/ContractManagerSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace ContractManager


        Public Class ContractManagerSQL
            Private _database As Sys.Database

            Public Sub New(ByVal database As Sys.Database)
                _database = database
            End Sub

#Region "Functions"
            Public Function Contracts_GetAll(ByVal datefrom As DateTime) As DataView
                _database.ClearParameter()
                _database.AddParameter("@ContractType1Enum", CInt(Entity.Sys.Enums.QuoteType.Contract_Opt_1), True)
                _database.AddParameter("@ContractType2Enum", CInt(Entity.Sys.Enums.QuoteType.Contract_Opt_2), True)
                _database.AddParameter("@ContractType3Enum", CInt(Entity.Sys.Enums.QuoteType.Contract_Opt_3), True)
                _database.AddParameter("@UploadStatusEnum", CInt(Entity.Sys.Enums.VisitStatus.Uploaded), True)
                _database.AddParameter("@DateFrom", datefrom, True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("Contracts_GetAll_Mk1")
                dt.TableName = Entity.Sys.Enums.TableNames.tblContract.ToString
                Return New DataView(dt)
            End Function

            Public Sub ContractRenewalsInsert(ByVal oldContractID As Integer, ByVal newContractID As Integer, ByVal ContractTypeID As Integer)
                _database.ClearParameter()
                _database.AddParameter("@oldContractID", oldContractID, True)
                _database.AddParameter("@newContractID", newContractID, True)
                _database.AddParameter("@ContractTypeID", ContractTypeID, True)
                DB.ExecuteSP_NO_Return("ContractRenewalsInsert")
            End Sub

            Public Function ContractRenewals_GetNewID_ByOldID(ByVal oldContractID As Integer, ByVal ContractTypeID As Integer) As Integer
                _database.ClearParameter()
                _database.AddParameter("@oldContractID", oldContractID, True)
                _database.AddParameter("@ContractTypeID", ContractTypeID, True)
                Return Entity.Sys.Helper.MakeIntegerValid(DB.ExecuteSP_OBJECT("ContractRenewals_GetNewID_ByOldID"))
            End Function

            Public Function ContractRenewal_AlternativeSitesJobOfWorks(ByVal ContractID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@ContractID", ContractID, True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("ContractRenewal_AlternativeSitesJobOfWorks")
                dt.TableName = Entity.Sys.Enums.TableNames.NO_TABLE.ToString
                Return New DataView(dt)
            End Function


            Public Function Contracts_GetAddresses(ByVal ContractIDs As String) As DataView
                _database.ClearParameter()
                _database.AddParameter("@ContractIDs", ContractIDs, True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("Contracts_GetAddresses")
                dt.TableName = Entity.Sys.Enums.TableNames.NO_TABLE.ToString
                Return New DataView(dt)
            End Function

#End Region
        End Class
    End Namespace
End Namespace

File: src/Application Specific/Classes/Entity Collections/ContractManager/ContractManagerSQL.cs
using System;
using System.Data;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace ContractManager
    {
        public class ContractManagerSQL
        {
            private Sys.Database _database;

            public ContractManagerSQL(Sys.Database database)
            {
                _database = database;
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public DataView Contracts_GetAll(DateTime datefrom)
            {
                _database.ClearParameter();
                _database.AddParameter("@ContractType1Enum", Conversions.ToInteger(Sys.Enums.QuoteType.Contract_Opt_1), true);
                _database.AddParameter("@ContractType2Enum", Conversions.ToInteger(Sys.Enums.QuoteType.Contract_Opt_2), true);
                _database.AddParameter("@ContractType3Enum", Conversions.ToInteger(Sys.Enums.QuoteType.Contract_Opt_3), true);
                _database.AddParameter("@UploadStatusEnum", Conversions.ToInteger(Sys.Enums.VisitStatus.Uploaded), true);
                _database.AddParameter("@DateFrom", datefrom, true);
                var dt = _database.ExecuteSP_DataTable("Contracts_GetAll_Mk1");
                dt.TableName = Sys.Enums.TableNames.tblContract.ToString();
                return new DataView(dt);
            }

            public void ContractRenewalsInsert(int oldContractID, int newContractID, int ContractTypeID)
            {
                _database.ClearParameter();
                _database.AddParameter("@oldContractID", oldContractID, true);
                _database.AddParameter("@newContractID", newContractID, true);
                _database.AddParameter("@ContractTypeID", ContractTypeID, true);
                App.DB.ExecuteSP_NO_Return("ContractRenewalsInsert");
            }

            public int ContractRenewals_GetNewID_ByOldID(int oldContractID, int ContractTypeID)
            {
                _database.ClearParameter();
                _database.AddParameter("@oldContractID", oldContractID, true);
                _database.AddParameter("@ContractTypeID", ContractTypeID, true);
                return Sys.Helper.MakeIntegerValid(App.DB.ExecuteSP_OBJECT("ContractRenewals_GetNewID_ByOldID"));
            }

            public DataView ContractRenewal_AlternativeSitesJobOfWorks(int ContractID)
            {
                _database.ClearParameter();
                _database.AddParameter("@ContractID", ContractID, true);
                var dt = _database.ExecuteSP_DataTable("ContractRenewal_AlternativeSitesJobOfWorks");
                dt.TableName = Sys.Enums.TableNames.NO_TABLE.ToString();
                return new DataView(dt);
            }

            public DataView Contracts_GetAddresses(string ContractIDs)
            {
                _database.ClearParameter();
                _database.AddParameter("@ContractIDs", ContractIDs, true);
                var dt = _database.ExecuteSP_DataTable("Contracts_GetAddresses");
                dt.TableName = Sys.Enums.TableNames.NO_TABLE.ToString();
                return new DataView(dt);
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}